mkosi/__init__.py

```diff
@@ -962,13 +979,39 @@ def gen_kernel_images(state: MkosiState) -> Iterator[tuple[str, Path]]:
         yield kver.name, kimg
 
 
-def initrd_path(state: MkosiState, kver: str) -> Path:
+def initrd_paths(state: MkosiState, kver: str) -> list[Path]:
+    if state.config.initrds:
+        kmods_initrd = kernel_modules_initrd_path(state, kver)
+        assert kmods_initrd
+        return [*state.config.initrds, kmods_initrd]
+
     # initrd file is versioned in Debian Bookworm
     initrd = state.root / boot_directory(state, kver) / f"initrd.img-{kver}"
     if not initrd.exists():
         initrd = state.root / boot_directory(state, kver) / "initrd"
 
-    return initrd
+    return [initrd]
+
+
+def kernel_modules_initrd_path(state: MkosiState, kver: str) -> Optional[Path]:
+    if not state.config.initrds:
+        return None
+
+    compression = kernel_modules_initrd_compression(state.config)
+    if compression == "xz":
+        ext = "xz"
+    elif compression == "zstd":
+        ext = "zst"
+    else:
+        die(f"Unknown compression {compression}")
+
+    return state.var_tmp() / f"initrd.{kver}.kmods.{ext}"
+
+
+def kernel_modules_initrd_cpio_path(state: MkosiState, kver: str) -> Optional[Path]:
+    if not state.config.initrds:
+        return None
+    return state.var_tmp() / f"initrd.{kver}.kmods"
```

Comment:
This should go into `state.workspace`, `state.var_tmp()` is for files generated by tools running in the context of the image